libbeat/docs/processors.asciidoc

```diff
@@ -25,3 +25,5 @@ order they are defined in the {beatname_uc} configuration file.
 -------
 event -> processor 1 -> event1 -> processor 2 -> event2 ...
 -------
+
+IMPORTANT: Please configure processors in a way that would not remove or overwrite fields created by {beatname_uc}. Certain processors might need these fields to function. If it's absolutely necessary to remove, rename or overwrite a pre-defined field, please make sure it's done by a corresponding processor (<<drop-fields,`drop_fields`>>, <<rename-fields,`rename`>> or <<add-fields, `add_fields`>>) placed at the end of the processor list defined in the input configuration.
```

Comment:
Do we have a list of the fields create by each Beat, or can we just list the fields specific processors might assume to exist?

This warning is very broad, if I was a user I wouldn't know if this applies to me.

Reply:
I agree with Craig. Is this change coming from a specific issue we faced/was reported?

Another option would be to suggest the user reads the documentation for the processor they want to use in order to find out the required fields.

On a quick glance I believe most processors act on configured fields, the [community ID](https://www.elastic.co/guide/en/beats/filebeat/current/community-id.html) is one of the few that stood out to me as expecting a set of fields to be present in the event.

Reply:
@cmacknz I don't think we should have a list like this. This statement is broad on purpose, so it does not limit us anyhow building processors or other logic around the existing event schema. Otherwise, we would have to:

1. Investigate what we're currently using in our code and processing all pipelines (don't forget about integrations)
2. Compose such a list
3. Version and maintain this list
4. Never break backwards compatibility.

Ideally our customers should not break the existing schema, only extend it. However, if they must change something (e.g. due a data protection), there is a described workaround in the docs.

Reply:
What about improving a bit the copy to explicitly mention consulting the used processor's documentation? So at least users have a clear direction of where to find the necessary information about which fields they can change or not.

Reply:
> What about improving a bit the copy to explicitly mention consulting the used processor's documentation?

@belimawr not sure I understand the question. You mean explaining in each processor what fields it's reading from the event in order to function? If so, then see my comment above https://github.com/elastic/beats/pull/37459#discussion_r1432528284

Reply:
> that would not remove or overwrite fields created by {beatname_uc}.

I think my main problem is this statement which could be interpreted to mean every single field. Which fields are not created by the Beat? 

I suspect you are trying to distinguish fields an input has generated itself vs fields read from the event but I don't think this is clear enough. We need to provide some specific examples as well.

Reply:
How about something like:

Elastic recommends doing all drop and renaming of existing fields as the last step in a processor configuration.  This is because dropping or renaming fields can remove data necessary for the next processor in the list, for example dropping the `source.ip` field would remove one of the fields necessary for the `community_id` processor to function.